include/i2s.h

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2017 Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for the I2S (Inter-IC Sound) bus drivers.
+ */
+
+#ifndef __I2S_H__
+#define __I2S_H__
+
+/**
+ * @defgroup i2s_interface I2S Interface
+ * @ingroup io_interfaces
+ * @brief I2S (Inter-IC Sound) Interface
+ *
+ * The I2S API provides support for the standard I2S interface standard as well
+ * as common non-standard extensions such as PCM Short/Long Frame Sync,
+ * Left/Right Justified Data Format.
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * The following #defines are used to configure the I2S controller.
+ */
+
+
+typedef u8_t i2s_fmt_t;
```

Comment:
I've used specific types for two reasons: 
1) it helps user to determine which #define applies to which parameter. 
2) it is likely that i2s_fmt_t and especially i2s_opt_t will have to occupy more than one byte in the future when new options are added. Having a dedicated type, rather than u8_t makes it easier to extend it when the need arises.